pkg/routeagent-driver/main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"flag"
+	"os"
+
+	"github.com/kelseyhightower/envconfig"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
+
+	"github.com/submariner-io/submariner/pkg/event"
+	"github.com/submariner-io/submariner/pkg/event/controller"
+	"github.com/submariner-io/submariner/pkg/event/logger"
+	kp "github.com/submariner-io/submariner/pkg/routeagent-driver/handlers/kubeproxy-iptables"
+)
+
+var (
+	masterURL  string
+	kubeconfig string
+)
+
+type specification struct {
+	ClusterID   string
+	Namespace   string
+	ClusterCidr []string
+	ServiceCidr []string
+}
+
+func main() {
+	klog.InitFlags(nil)
+	flag.Parse()
+
+	klog.Info("SGM: Starting submariner-route-agent using the event framework")
+	// set up signals so we handle the first shutdown signal gracefully
+	stopCh := signals.SetupSignalHandler()
+
+	var env specification
+	err := envconfig.Process("submariner", &env)
+	if err != nil {
+		klog.Fatalf("Error reading the environment variables: %s", err.Error())
+	}
+
+	eventHandlers := event.NewRegistry("routeagent-driver", os.Getenv("NETWORK_PLUGIN"))
+	if err := eventHandlers.AddHandlers(logger.NewHandler(),
+		kp.NewSyncHandler(env.ClusterID, env.Namespace, env.ClusterCidr, env.ServiceCidr)); err != nil {
```

Comment:
nit: You could just pass the env object, but you may need to move the env variable specification into a common module

Reply:
Yes I thought about it, but wanted to have explicit arguments as it will be clear on what we are passing to the SyncHandler.